core/domain/rights_manager.py

```diff
@@ -396,60 +581,155 @@ def _can_delete(self, rights_object):
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
 
-        This is true if the activity is community-owned, or if the user is in
-        the owner/editor list for the activity.
+        Returns:
+            bool. Whether the user is in the owner/editor list for the activity.
         """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._has_editing_rights(activity_rights)
 
     def has_viewing_rights(self, activity_type, activity_id):
+        """Returns whether this user has viewing rights for given activity.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to view an activity.
+        """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._has_viewing_rights(activity_rights)
 
     def can_play(self, activity_type, activity_id):
-        """Whether the user can play the reader view of this activity."""
+        """Returns whether this user has rights to play a given activity.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to play an activity.
+        """
         activity_rights = _get_activity_rights(activity_type, activity_id)
         if activity_rights is None:
             return False
         return self._can_play(activity_rights)
 
     def can_view(self, activity_type, activity_id):
-        """Whether the user can view the editor page for this activity."""
+        """Returns whether this user has rights to view a given activity.
+
+        Args:
+            activity_type: The type of activity. Either one of these:
+                feconf.ACTIVITY_TYPE_EXPLORATION
+                feconf.ACTIVITY_TYPE_COLLECTION
+            activity_id: str. ID of activity.
+
+        Returns:
+            bool. Whether the user has a right to view an activity.
+        """
         return self.can_play(activity_type, activity_id)
 
     def can_edit(self, activity_type, activity_id):
-        # TODO(sll): Add a check here for whether a user is banned or not,
-        # rather than having this check in the controller.
+        """Returns whether this user has rights to edit a given activity.
+
+        TODO(sll): Add a check here for whether a user is banned or not,
```

Comment:
Could we put this above line 680 (outside the docstring)? Docstrings shouldn't have TODOs in them.